File: java/src/main/java/com/sasteval/vuln/easy/ErrorInfoLeakDirect.java
package com.sasteval.vuln.easy;

import com.sasteval.util.DbUtil;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.sql.Connection;
import java.sql.Statement;

/**
 * CWE-209: Information Exposure Through an Error Message (Direct)
 * VULNERABILITY: Full stack trace is written to HTTP response.
 */
public class ErrorInfoLeakDirect extends HttpServlet {

    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {

        String id = request.getParameter("id");
        response.setContentType("text/plain");

        try {
            Connection conn = DbUtil.getConnection();
            Statement stmt = conn.createStatement();
            stmt.executeQuery("SELECT * FROM users WHERE id = " + id);
        } catch (Exception e) {
            // VULN: Full stack trace sent to client, leaking internal details
            e.printStackTrace(response.getWriter());
        }
    }
}

File: dotnet/Vuln/Easy/ErrorInfoLeakDirect.cs
using System.Data.SQLite;
using SastEval.Util;

namespace SastEval.Vuln.Easy;

/// <summary>
/// CWE-209: Information Exposure Through an Error Message - Full stack trace returned to client.
/// </summary>
public static class ErrorInfoLeakDirect
{
    public static IResult Handle(HttpContext context)
    {
        var id = context.Request.Query["id"].ToString();

        try
        {
            var conn = DbUtil.GetConnection();
            using var cmd = new SQLiteCommand("SELECT * FROM users WHERE id = " + id, conn);
            using var reader = cmd.ExecuteReader();
            return Results.Text("Query executed");
        }
        catch (Exception ex)
        {
            // VULNERABLE: Full exception details including stack trace returned to user
            return Results.Text(ex.ToString());
        }
    }
}
